src/storage/src/hummock/store/hummock_storage.rs

```diff
@@ -416,6 +429,14 @@ impl StateStoreRead for HummockStorage {
         epoch: u64,
         read_options: ReadOptions,
     ) -> impl Future<Output = StorageResult<Self::IterStream>> + '_ {
+        let (l_vnode_inclusive, r_vnode_exclusive) = vnode_range(&key_range);
```

Comment:
How about making it unified, moving assert logic to functions `vnode` and `vnode_part`.

I'm not sure whether we need to strictly enforce checks on every call?

Reply:
> How about making it unified, moving assert logic to functions vnode and vnode_part.
`fn vnode(range)` also have the same assertion. Given that the cost of the assertion is small, I would rather make the assertion more explicit by putting in at the beginning of all the `iter` calls.

> I'm not sure whether we need to strictly enforce checks on every call?

This is the main motivation of this PR to avoid misuage of storage iter.